File: Obj-c/SampleLoginMeuID/SampleLoginMeuID/Success/SuccessViewController.m
//
//  SuccessViewController.m
//  SampleLoginMeuID
//
//  Created by Paulo Rosa on 08/12/20.
//  Copyright © 2020 Idwall. All rights reserved.
//

#import "SuccessViewController.h"

@interface SuccessViewController ()

@end

@implementation SuccessViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (void)setParametersFromURL:(NSURL *)url {
    NSString *code = [self queryParametersFromURL:url][@"code"];
    NSString *codeVerifier = [self queryParametersFromURL:url][@"code_verifier"];
    self.codeLabel.text = [_codeLabel.text stringByAppendingString:code];
    self.codeVerifierLabel.text = [_codeVerifierLabel.text stringByAppendingString:codeVerifier];
}

- (NSDictionary<NSString *, NSString *> *)queryParametersFromURL:(NSURL *)url {
    NSURLComponents *urlComponents = [NSURLComponents componentsWithURL:url resolvingAgainstBaseURL:NO];
    NSMutableDictionary<NSString *, NSString *> *queryParams = [NSMutableDictionary<NSString *, NSString *> new];
    for (NSURLQueryItem *queryItem in [urlComponents queryItems]) {
        if (queryItem.value == nil) { continue; }
        [queryParams setObject:queryItem.value forKey:queryItem.name];
    }
    return queryParams;
}
@end

File: Swift/SampleLoginMeuID/SampleLoginMeuID/Success/SuccessViewController.swift
//
//  SuccessViewController.swift
//  SampleLoginMeuID
//
//  Created by Paulo Rosa on 08/12/20.
//  Copyright © 2020 Idwall. All rights reserved.
//

import UIKit

class SuccessViewController: UIViewController {
   
    // MARK: - Outlets
    @IBOutlet weak var codeLabel: UILabel!
    @IBOutlet weak var codeVerifierLabel: UILabel!
    
    // MARK: - Properties
    let codeString = "code"
    let codeVerifierString = "code_verifier"
    
    // MARK: - Life Cycle
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    // MARK: - Public Functions
    func setParametersFromURL(url: URL) {
        if let codeParam = getQueryStringParameter(url: url.absoluteString, param: codeString),
            let codeVerifierParam = getQueryStringParameter(url: url.absoluteString, param: codeVerifierString) {
            codeLabel.text = "code: \(codeParam)"
            codeVerifierLabel.text = "code_verifier: \(codeVerifierParam)"
        }
    }
    
    // MARK: - Private Functions
    private func getQueryStringParameter(url: String, param: String) -> String? {
        guard let url = URLComponents(string: url) else { return nil }
        return url.queryItems?.first(where: { $0.name == param })?.value
    }
}
